requirements/requirements.txt

```diff
@@ -1,7 +1,12 @@
+hatchling 
 huggingface_hub>=0.24
+llama-index==0.10.43
```

Comment:
Why are these being added to core dependencies ?
Put them in another file and add it to setup.py. 
Core requirements are only for that which all domains require - llama-index is not used by all domains.

Reply:
Do you have a suggestion for a better requirements txt? I added all into the top-level file since I'm not familiar with dependencies of our collections.